client/keys.go

```diff
@@ -54,11 +62,34 @@ func NewDiscoveryKeysAPI(c HTTPClient) KeysAPI {
 }
 
 type KeysAPI interface {
-	Create(ctx context.Context, key, value string, ttl time.Duration) (*Response, error)
+	Set(ctx context.Context, key, value string, opts *SetOptions) (*Response, error)
+	Create(ctx context.Context, key, value string) (*Response, error)
```

Comment:
To expand on that - we're making a tradeoff here between readability and functionality. The common case is to Create/Update a key without a TTL. We don't want to force users to set a dummy value for the TTL, then have to question what it means every time they read the code. For example, what do the `-1` or `0` mean below?

```
Create(ctx, "/foo", "bar", -1)
Update(ctx, "/foo", "baz", 0)
```

If you care about setting the TTL or doing any comparisons, define them in a SetOptions struct and use the Set method. You could do a Create with a TTL through Set with `SetOptions{PrevExist: false, TTL: time.Minute}`